editor/src/components/canvas/canvas-strategies/strategies/keyboard-absolute-resize-strategy.tsx

```diff
@@ -111,6 +117,97 @@ function getFitness(interactionSession: InteractionSession | null): number {
 
 export const ResizeMinimumValue = 1
 
+export interface ChangeBoundsResult {
+  commands: Array<CanvasCommand>
+  intendedBounds: Array<CanvasFrameAndTarget>
+}
+
+export function changeBounds(
+  projectContents: ProjectContentTreeRoot,
+  startingMetadata: ElementInstanceMetadataMap,
+  selectedElements: Array<ElementPath>,
+  originalFrame: CanvasRectangle,
+  originalIntendedBounds: Array<CanvasFrameAndTarget>,
+  edgePosition: EdgePosition,
+  movement: CanvasVector,
+): ChangeBoundsResult {
+  let commands: Array<CanvasCommand> = []
+  let intendedBounds: Array<CanvasFrameAndTarget> = originalIntendedBounds
+
+  selectedElements.forEach((selectedElement) => {
+    const element = withUnderlyingTarget(selectedElement, projectContents, null, (_, e) => e)
```

Comment:
`getElementFromProjectContents`

Reply:
This was extracted out from its original location.

Reply:
Be careful with `getElementFromProjectContents`, it only returns an element if it is a JSXElement (I'm changing that in my in-progress PR)

Reply:
this is just moved code AFAIK